yt/geometry/grid_geometry_handler.py

```diff
@@ -106,6 +106,27 @@ def clear_all_data(self):
             g.clear_data()
         self.io.queue.clear()
 
+    def set_grid_cache_mask(self, cache_grid_masks, clear_all_data=True):
```

Comment:
I'm slightly unsure about this being a top-level method, rather than having an underscore prefix.  Then again, maybe it's good.

Reply:
Ya, I had the same thought. I think an underscore might be better? It's probably not something 99.9% of people would use. But maybe someone might find themselves in a memory-limited situation and want to disable cacheing easily? 

So ya, I was on the fence, but happy to go with an underscore instead.

Reply:
Let's do underscore, then I can merge and we'll PR upstream

Reply:
done